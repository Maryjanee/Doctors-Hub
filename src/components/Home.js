import { Link } from 'react-router-dom';

const Home = () => (
  <div>
    <p>Home Page</p>
    <Link to="/login">Login</Link>
    <br />
    <Link to="/signup">Sign Up</Link>
  </div>
);
export default Home;
