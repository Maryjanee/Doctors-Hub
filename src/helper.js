/* eslint-disable import/prefer-default-export */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable arrow-body-style */
import { BrowserRouter } from 'react-router-dom';
import { createStore, applyMiddleware } from 'redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import rootReducer from './reducers';

export const renderComponent = (Component, props = {}) => {
  return (
    <BrowserRouter>
      <Component {...props} />
    </BrowserRouter>
  );
};

export const renderWithRedux = (component,
  {
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  } = {}) => ({
  ...render(
    <Provider store={store}>
      renderComponent(component)
    </Provider>,
  ),
});
