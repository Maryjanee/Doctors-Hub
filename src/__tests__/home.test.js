import renderer from 'react-test-renderer';
import Home from '../components/Home';
import { renderComponent } from '../helper';

describe('Home Page', () => {
  it('deep snapshot renders correctly', () => {
    const component = renderComponent(Home);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
