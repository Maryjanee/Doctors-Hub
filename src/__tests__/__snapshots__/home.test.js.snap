// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page deep snapshot renders correctly 1`] = `
<div>
  <p>
    Home Page
  </p>
  <a
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
  <br />
  <a
    href="/signup"
    onClick={[Function]}
  >
    Sign Up
  </a>
</div>
`;
